pkg/reconciler/kubernetes/tektonpipelinesascode/controller.go

```diff
@@ -0,0 +1,117 @@
+	tpacreconciler "github.com/tektoncd/operator/pkg/client/injection/reconciler/operator/v1alpha1/tektonpipelinesascode"
+	"github.com/tektoncd/operator/pkg/reconciler/common"
+	"go.uber.org/zap"
+	"k8s.io/client-go/tools/cache"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/injection"
+	"knative.dev/pkg/logging"
+)
+
+// NewController initializes the controller and is called by the generated code
+// Registers eventhandlers to enqueue events
+func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	return NewExtendedController(common.NoExtension)(ctx, cmw)
+}
+
+// NewExtendedController returns a controller extended to a specific platform
+func NewExtendedController(generator common.ExtensionGenerator) injection.ControllerConstructor {
+	return func(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+		logger := logging.FromContext(ctx)
+
+		mfclient, err := mfc.NewClient(injection.GetConfig(ctx))
+		if err != nil {
+			logger.Fatalw("Error creating client from injected config", zap.Error(err))
+		}
+		mflogger := zapr.NewLogger(logger.Named("manifestival").Desugar())
+		manifest, err := mf.ManifestFrom(mf.Slice{}, mf.UseClient(mfclient), mf.UseLogger(mflogger))
+		if err != nil {
+			logger.Fatalw("Error creating initial manifest", zap.Error(err))
+		}
+
+		if err := fetchSourceManifests(ctx, &manifest); err != nil {
+			logger.Fatalw("failed to read manifest", err)
+		}
+
+		version, err := fetchVersion(manifest)
+		if err != nil {
+			logger.Fatal("Failed to find version from env")
+		}
+
+		c := &Reconciler{
+			operatorClientSet: operatorclient.Get(ctx),
+			extension:         generator(ctx),
+			pipelineInformer:  tektonPipelineinformer.Get(ctx),
+			triggerInformer:   tektonTriggerinformer.Get(ctx),
+			manifest:          manifest,
+			version:           version,
+		}
+		impl := tpacreconciler.NewImpl(ctx, c)
+
+		// Add enqueue func in reconciler
+		c.enqueueAfter = impl.EnqueueAfter
+
+		logger.Info("Setting up event handlers for TektonPipelinesAsCode")
+
+		tpacinformer.Get(ctx).Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
+
+		tektonInstallerinformer.Get(ctx).Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+			FilterFunc: controller.FilterController(&v1alpha1.TektonPipelinesAsCode{}),
+			Handler:    controller.HandleAll(impl.EnqueueControllerOf),
+		})
+
+		return impl
+	}
+}
+
+func fetchSourceManifests(ctx context.Context, manifest *mf.Manifest) error {
+	var pac *v1alpha1.TektonPipelinesAsCode
+	return common.AppendTarget(ctx, manifest, pac)
+}
+
+func fetchVersion(manifest mf.Manifest) (string, error) {
+	crds := manifest.Filter(mf.CRDs)
+
+	if len(crds.Resources()) == 0 {
+		return "", fmt.Errorf("failed to find CRD in source manifest")
+	}
+
+	labels := crds.Resources()[0].GetLabels()
+	version, ok := labels["app.kubernetes.io/version"]
```

Comment:
may be we should create an issue in PAC to create an "info" configMap which holds version information, like other components.

https://github.com/tektoncd/triggers/blob/d360e8f7e51ca97d4cc1fc072c955a5365a865f6/config/config-info.yaml#L29